Bottom: 4fa2d2eb3eeeff4c1dc33c67228929fcfd37b5cf
Top:    437b957f9b9b33615925d25b01225db9c40789c0
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2013-08-02 15:00:58 -0700

Refresh of rs-setroute

---

diff --git a/src/rsocket.c b/src/rsocket.c
index d64c3bb..57df290 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -1251,15 +1251,20 @@ resolve_route:
 					      rs->optlen);
 			free(rs->optval);
 			rs->optval = NULL;
+			if (!ret) {
+				rs->state = rs_resolving_route;
+				goto resolving_route;
+			}
 		} else {
 			ret = rdma_resolve_route(rs->cm_id, to);
+			if (!ret)
+				goto do_connect;
 		}
-		if (!ret)
-			goto do_connect;
 		if (errno == EAGAIN || errno == EWOULDBLOCK)
 			rs->state = rs_resolving_route;
 		break;
 	case rs_resolving_route:
+resolving_route:
 		ret = ucma_complete(rs->cm_id);
 		if (ret) {
 			if (errno == ETIMEDOUT && rs->retries <= RS_CONN_RETRIES)
